lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -548,6 +553,92 @@ def _graded_scorable_blocks_to_header(cls, course):
 
 
 class ProblemResponses(object):
+
+    @classmethod
+    def _build_problem_list(cls, course_blocks, root, path=None):
+        """
+        Generate a tuple of display names, block location paths and block keys
+        for all problem blocks under the ``root`` block.
+
+        Arguments:
+            course_blocks (BlockStructureBlockData): Block structure for a course.
+            root (UsageKey): This block and its children will be used to generate
+                the problem list
+            path (List[str]): The list of display names for the parent of root block
+
+        Yields:
+            Tuple[str, List[str], UsageKey]: tuple of a block's display name, path, and
+                usage key
+        """
+        display_name = course_blocks.get_xblock_field(root, 'display_name')
+        if path is None:
+            path = [display_name]
+
+        yield display_name, path, root
+
+        for block in course_blocks.get_children(root):
+            display_name = course_blocks.get_xblock_field(block, 'display_name')
+            for result in cls._build_problem_list(course_blocks, block, path + [display_name]):
+                yield result
+
+    @classmethod
+    def _build_student_data(cls, user_id, course_id, problem_location):
+        """
+        Generate a list of problem responses for all problem under the
+        ``problem_location`` root.
+
+        Arguments:
+            user_id (int): The user id for the user generating the report
+            course_id (CourseKey): The ``CourseKey`` for the course whose report
+                is being generated
+            problem_location (str): The generated report will include this
```

Comment:
Would be a lot more consistent to call them `course_key` and `usage_key_string`.